irods-vfs-impl/src/main/java/org/irods/nfsrods/vfs/IRODSVirtualFileSystem.java

```diff
@@ -1215,9 +1223,14 @@ else if (srcFile.isDirectory())
 
             log_.debug("move - Updating mappings between paths and inodes ...");
 
-            // If the destination path exists in iRODS and has been mapped by the NFSRODS
-            // server, then that path must be unmapped before remapping to avoid an exception.
-            if (unmapDstPath)
+            // If the destination path represents a data object, unmap it. This reflects
+            // the fact that if both the source and destination represent files, the destination
+            // object will no longer exist.
+            //
+            // If the destination path represents an empty collection, unmap it. This reflects
+            // the behavior seen in Linux. That is, if the destination directory is empty, it is
+            // renamed, otherwise the source is moved into the destination directory.
+            if (dstFileIsDataObject || dstFileIsEmpty)
```

Comment:
I think I got this wrong.

We should only unmap files/data-objects, NOT directories/collections.

iRODS does not allow collections to be overwritten so `mv -T` is not supported here.

Reply:
Interesting.    Collections are just virtual... so that does make sense I guess.

Reply:
Rerunning the tests now. Should be done in roughly 30 mins.

Reply:
All tests passed.